lib/textstat.rb

```diff
@@ -95,37 +93,35 @@ def self.smog_index(text)
     if sentences >= 3
       begin
         polysyllab = polysyllab_count(text)
-        smog = (
-        (1.043 * (30 * (polysyllab / sentences))**0.5) + 3.1291)
-        return smog.round(1)
+        smog = 1.043 * Math.sqrt(30.0 * polysyllab / sentences) + 3.1291
+        smog.round(1)
       rescue ZeroDivisionError
-        return 0.0
+        0.0
       end
     else
-      return 0.0
+      0.0
     end
   end
 
   def self.coleman_liau_index(text)
-    letters   = (avg_letter_per_word(text) * 100).round(2)
+    letters = (avg_letter_per_word(text) * 100).round(2)
     sentences = (avg_sentence_per_word(text) * 100).round(2)
-    coleman   = ((0.058 * letters) - (0.296 * sentences) - 15.8).to_f
+    coleman = 0.0588 * letters - 0.296 * sentences - 15.8
     coleman.round(2)
   end
 
   def self.automated_readability_index(text)
-    chars     = char_count(text)
-    words     = lexicon_count(text)
+    chars = char_count(text)
+    words = lexicon_count(text)
     sentences = sentence_count(text)
     begin
-      a = chars.to_f / words.to_f
-      b = words.to_f / sentences.to_f
+      a = chars.to_f / words
+      b = words.to_f / sentences
 
-      readability = (
-      (4.71 * a.round(2) + (0.5 * b.round(2))) - 21.43)
```

Comment:
not sure why these were being rounded mid-calculation? I couldn't find any material suggesting that this was a part of the calculation?!